lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module

```diff
@@ -127,14 +127,18 @@ function dosomething_organ_donation_send_postal_code() {
  *
  */
 function dosomething_organ_donation_store_registration() {
+  // Grab this to get the sid
   $params = drupal_get_query_parameters();
 
+  // Grab this to get the uid
+  global $user;
+
   try {
-    if ($params['sid'] && $params['uid']) {
+    if ($params['sid'] && $user->uid) {
```

Comment:
@sbsmith86 I added that into this PR, can you 👀 with the latest commit? thank you!